app/models/namespace.rb

```diff
@@ -225,4 +225,32 @@ def validate_nesting_level
   def restore_routes
     Route.restore(Route.only_deleted.find_by(source_id: id).id, recursive: true)
   end
+
+  def subtract_from_metadata_summary(namespaces, metadata, update_by_one)
+    namespaces.each do |namespace|
+      metadata.each do |metadata_field, value|
+        value = 1 if update_by_one
+        if namespace.metadata_summary[metadata_field] == value
+          namespace.metadata_summary.delete(metadata_field)
+        else
+          namespace.metadata_summary[metadata_field] -= value
+        end
+      end
+      namespace.save
+    end
+  end
+
+  def add_to_metadata_summary(namespaces, metadata, update_by_one)
+    namespaces.each do |namespace|
+      metadata.each do |metadata_field, value|
+        value = 1 if update_by_one
+        if namespace.metadata_summary.key?(metadata_field)
+          namespace.metadata_summary[metadata_field] += value
```

Comment:
Question: Same as above comment. Is it possible for update_by_one to be false and the value not an integer?

Reply:
See above!